src/platform/packages/shared/kbn-server-route-repository/src/test_types.ts

```diff
@@ -228,6 +270,55 @@ const repository = {
       return of({ type: 'foo' as const, streamed_response: true });
     },
   }),
+  ...createServerRoute({
+    endpoint: 'GET /internal/endpoint_with_response_validation_zod',
+    params: z.object({
+      query: z.object({
+        start: z.string(),
+      }),
+    }),
+    responseValidation: {
+      200: {
+        body: z.object({
+          success: z.literal(true),
+          data: z.array(z.object({ id: z.number() })),
+        }),
+      },
+      202: {
+        body: z.object({
+          success: z.literal(true),
+          data: z.array(z.object({ id: z.number() })).length(0),
+        }),
+      },
+      400: {
+        body: z.object({
+          success: z.literal(false),
+          error: z.string(),
+        }),
+      },
+    },
+    async handler(resources) {
+      const start = resources.params.query.start;
+
+      if (start === 'something-1') {
+        return {
+          success: true as const,
+          data: [{ id: 1 }, { id: 2 }],
+        };
+      }
+
+      if (start === 'something-2') {
+        return kibanaResponseFactory.accepted({
+          body: {
+            success: true as const,
+            data: [{ id: 1 }, { id: 2 }],
+          },
+        });
+      }
+
+      return { success: false as const, error: 'Example error' };
```

Comment:
Since there is no code changes done in register routes, this will result in a HTTP 200 OK, so I don't think the 400 response schema will be used, what we should do here is return a 400 using the response factory.